### PR TITLE
Dont allow smartchart to save duplicate templates

### PR DESCRIPTION
**Pull request by MahboobehMohammadi**

### CODE REVIEW

File: sass/components/_view.scss [hunk 1]
```diff
@@ -86,6 +87,50 @@
 			}
 		}
 	}
+	.cq-view-dialog
+	{
+		margin-top: 100;
```
Comment (brucebinary) [inline sass/components/_view.scss]: this doesn't do anything

File: sass/components/_view.scss [hunk 2]
```diff
@@ -86,6 +87,50 @@
 			}
 		}
 	}
+	.cq-view-dialog
```
Comment (brucebinary) [inline sass/components/_view.scss]: maybe this should be called `.cq-view-prompt`? Dialog doesn't seem that appropriate a name to use.

Also you are missing that tiny downward pointing arrow below the overwrite button.

File: src/store/DialogStore.js [hunk 3]
```diff
@@ -10,16 +10,18 @@ export default class DialogStore {
 
     @observable open = false;
 
-    @action.bound setOpen(val) {
+    @action.bound setOpen(val , closeActiveDialog = true) {
```
Comment (brucebinary) [inline src/store/DialogStore.js]: Is `closeActiveDialog` just for that overwrite prompt? Doesn't look very appropriate. That prompt should be a dialog by itself.

File: src/store/ViewStore.js [hunk 4]
```diff
@@ -1,11 +1,13 @@
 import { observable, action, when } from 'mobx';
 import { createObjectFromLocalStorage } from '../utils';
 import MenuStore from './MenuStore';
+import DialogStore from './DialogStore';
 
 export default class ViewStore {
     constructor(mainStore) {
         this.mainStore = mainStore;
         this.menu = new MenuStore(mainStore);
+        this.dialog = new DialogStore(mainStore);
```
Comment (brucebinary) [inline src/store/ViewStore.js]: how about `this.overwritePrompt`?

File: src/store/ViewStore.js [hunk 5]
```diff
@@ -46,13 +48,29 @@ export default class ViewStore {
...
+            this.updateRoute('main');
             const layout = this.stx.exportLayout();
             this.views.push({ name: this.templateName, layout });
             this.updateLocalStorage();
+            this.templateName = '';
+        }
+    }
+
+    @action.bound onOverwrite() {
+        if (this.templateName.length > 0) {
```
Comment (brucebinary) [inline src/store/ViewStore.js]: it's odd that we need this. User should not be able to change the input of the template or add anymore templates until they have dealt with the prompt.

File: src/components/Views.jsx [hunk 6]
```diff
@@ -40,8 +46,29 @@ const Views = ({
             />
         </Menu.Title>
         <Menu.Body>
+            <Dialog className="cq-dialog cq-view-dialog">
```
Comment (brucebinary) [inline src/components/Views.jsx]: Placing this inside `<Menu.Body>` resizes the menu dropdown when the prompt shows up. The dropdown and the dialog needs to be side by side.

File: sass/components/_view.scss [hunk 7]
```diff
@@ -86,6 +87,50 @@
...
+		padding-top: 10px;
+		display: flex;
+		justify-content:center;
+		>.ic-icon
+		{
+			-moz-transition: none;
+			-webkit-transition: none;
+			-ms-transition: none;
+    		-o-transition: color 0 ease-in;
+			transition: none;
```
Comment (brucebinary) [inline sass/components/_view.scss]: You don't need to write so much stuff; use the transition mixin defined in `_variables.scss`

File: src/store/DialogStore.js (hunk 3 above)
Comment (MahboobehMohammadi) [reply src/store/DialogStore.js]: when you open a dilaog it closes the dialog that is alrady opened, i want to open a dialog iniside another, so i tell it not to close the old one(the parent dialog here) , so this propery is basicly tell to not close the activedilaog.

Comment (brucebinary) [reply src/store/DialogStore.js]: aite. Fair point. I'm not sure if this meant we need to keep track of all the active dialogs if we have this setting... Just for one feature it's tedious though.

Comment (MahboobehMohammadi) [reply src/store/DialogStore.js]: no for the dialog you dont want want  keep track off. like confrim dialogs, in this case if the parrent closes the child that is the confrim dialog will be closed ass well. no need to track it.

File: src/store/DialogStore.js [hunk 8]
```diff
@@ -10,16 +10,18 @@ export default class DialogStore {
 
     @observable open = false;
 
-    @action.bound setOpen(val) {
+    @action.bound setOpen(val , isParentDialog = true) {
```
Comment (brucebinary) [inline src/store/DialogStore.js]: I like this name better (:

File: translation/messages.pot [hunk 9]
```diff
@@ -317,971 +326,977 @@ msgstr ""
 
 # Indicator strings:
 
-msgid "RSI"
+msgid "RSI"
```
Comment (brucebinary) [inline translation/messages.pot]: Why is there `^M` here...? These are Indicator strings and practically doesn't change. If you don't want to debug the `yarn translation` code (I suspect it's a windows issue), just retain the translation strings you want and discard the rest.

File: sass/_themes.scss [hunk 10]
```diff
@@ -32,6 +32,7 @@ $themes: (
     controlGrid:            #fafafb,
     controlDialogHead:      #FFFFFF,
     controlDialogBody:      #FAFAFA,
+    controlDialogButton:    #2a3052,
```
Comment (brucebinary) [inline sass/_themes.scss]: Another colour just for this button? We need to get our designers to stick to a certain palette :/ 

Can you use `activeColor` and see if Afiq is ok with that? It's same colour as the modal dialog buttons.

File: src/store/ViewStore.js [hunk 11]
```diff
@@ -78,6 +101,9 @@ export default class ViewStore {
     }
 
     inputRef = (ref) => {
-        if (ref) { ref.focus(); }
+        if (ref) {
+            ref.value = this.templateName;
```
Comment (brucebinary) [inline src/store/ViewStore.js]: you should not need to do this. Consider using a controlled component https://reactjs.org/docs/forms.html#controlled-components

File: src/store/ViewStore.js [hunk 12]
```diff
@@ -78,6 +101,9 @@ export default class ViewStore {
...
+            ref.focus();
```
Comment (brucebinary) [inline src/store/ViewStore.js]: if `focus` is only called during rerenders, you mind as well use `autoFocus` and eliminate needing to use `ref`.

**Pull request merged**